docs/home-automation/home-assistant/supervisor-addons/android-debug-bridge/adb.md

```diff
@@ -0,0 +1,336 @@
+# ADB - Android Debug Bridge
+
+Requirements: 
+
+ * [ADB Addon](https://github.com/hassio-addons/addon-adb/blob/v0.6.1/README.md)
+
+This guide walks you though how to configure Android Debug Bridge for controlling Android TV through [Home Assistant](https://www.home-assistant.io/)
+
+Initially you will need to install the add-on through the Home Assistant Supervisor Add-on Store. Once the Add-on is installed you will need to configure your devices as per the example below. 
+
+ ```Config
+    devices:
+    - 1.2.3.4
+    - 4.3.2.1
+    reconnect_timeout: 30
+    log_level: info
+    keys_path: /config/.androidkeys
+```
+
+**I would recommend you create a path in you config folder for the `.androidkeys` to be stored as this will help you in the future should you ever need to move your instance.**
+
+Once the configuration is in place start the add-on and check the logs for any errors. 
+
+## Configuring your Android TV 
+
+In this example I will be using the Shield TV to enable Network debugging. 
+
+1. Enable developer options - On you Android TV, go to `Settings > About SHIELD`, scroll to the bottom and tap Build Number several times.
+2. Enable network debugging - On you device in `Settings > Developer Options` enable Network Debugging
+3. Add the following configuration to your home assistant configuration
+
+```YAML
+media_player:
+
+  - platform: androidtv
+    host: 1.2.3.4
+    name: "Living Room Super Shield"
+    adb_server_ip: 127.0.0.1
+    adb_server_port: 5037
+```
+4. Restart Home Assistant
+5. Restart `ADB` in Home Assistant
+6. Allow `Home Assistant` to access to your device, remember the check the box to always remember this device
+
+
+## Connect to the container running ADB
+
+If you are using Home Assistant backed Ubuntu or other debian based OS this next step is done in Bash, if you are running the enclosed Home Assistant OS install the [Terminal & SSH plugin](https://github.com/home-assistant/hassio-addons/tree/master/ssh)
+
+Open you command line `Shell` to bash into the container
+
+    docker exec -it addon_a0d7b954_adb /bin/bash
```

Comment:
The container name in this example looks a bit odd, and possibly specific to your machine. Is there a canonical name for it / some way to find out its name?

Reply:
All the home assistant add-ons have the same name

Reply:
Including the hash on the end? Interesting

Reply:
`addon_a0d7b954_adb` will be the same on all instances

Reply:
![image](https://user-images.githubusercontent.com/15075388/79054498-2c853c80-7c3d-11ea-94d4-69176a828a4e.png)

two separate instances on separate hosts